app/models/manageiq/providers/google/cloud_manager/refresh_parser.rb

```diff
@@ -41,9 +41,12 @@ def get_zones
       end
 
       def get_flavors
-        # connection.flavors returns a duplicate flavor for every zone
-        # so build a unique list of flavors using the flavor id
-        flavors = @connection.flavors.to_a.uniq(&:id)
+        # Google API returns a duplicate flavor for every zone
+        # so build a unique list of flavors using the flavor
+        flavors = @connection.list_aggregated_machine_types.items.values.each_with_object([]) do |zone, arr|
+          arr.concat(zone.machine_types) if zone.machine_types
+        end
+        flavors.uniq!(&:id)
```

Comment:
Think this can be a bit simpler using flat_map+compact+uniq, how about:
```
flavors_by_zone = @connection.list_aggregated_machine_types.items
flavors = flavors_by_zone.values.flat_map(&:machine_types).compact.uniq(&:id)
```